src/polyrith.lean

```diff
@@ -234,16 +256,22 @@ do
   -- trace $ get_var_names m, 
   -- trace (polys.map poly.mk_string),
   -- trace p.mk_string,
-  sage_out ← sage_output [to_string R, (get_var_names m).to_string, (polys.map poly.mk_string).to_string, p.mk_string],
-  trace sage_out,
+  let args := [to_string R, (get_var_names m).to_string, (polys.map poly.mk_string).to_string, p.mk_string],
+  let args := to_string (is_trace_enabled_for `polyrith) :: args,
+  sage_out ← sage_output args,
+  if is_trace_enabled_for `polyrith then 
+    trace sage_out
+  else do 
   coeffs_as_poly ← convert_sage_output sage_out,
   coeffs_as_pexpr ← coeffs_as_poly.mmap (poly.to_pexpr m),
   let eq_names := eq_names.map expr.local_pp_name,  
   coeffs_as_expr ← coeffs_as_pexpr.mmap $ λ e, to_expr ``(%%e : %%R),
   -- trace coeffs_as_expr,
-  expr_string ← (eq_names.zip coeffs_as_expr).mfoldl (λ s p, do ps ← tactic.pp p, return $ s ++ " " ++ to_string ps) "",
   linear_combo.linear_combination eq_names coeffs_as_pexpr,
-  trace!"Try this: linear_combination{expr_string}"
+  let components := (eq_names.zip coeffs_as_expr).filter $ λ pr, bnot $ pr.2.is_app_of `has_zero.zero,
+  expr_string ← components.mfoldl (λ s p, do ps ← tactic.pp p, return $ s ++ format.line ++ ps) "",
```

Comment:
I'm also using a `format` object instead of a string, which gives some more control over line breaks in the output.